src/axom/inlet/tests/inlet_Inlet.cpp

```diff
@@ -451,6 +451,105 @@ TEST(inlet_Inlet_basic, getNestedValuesAddedUsingTable)
 
 }
 
+TEST(inlet_Inlet, mixLevelTables)
+{
+  // regression test for the lua stack being at the wrong
+  // level with nested tables with non-nested values
+
+  axom::sidre::DataStore dataStore;
+  auto luaReader = std::make_shared<axom::inlet::LuaReader>();
+  luaReader->parseString(
+    "thermal_solver={"
+    "   u0 = { type = \"function\", func = \"BoundaryTemperature\"},"
+    "   kappa = { type = \"constant\", constant = 0.5},"
+    "   solver = {"
+    "     rel_tol = 1.e-6,"
+    "     abs_tol = 1.e-12,"
+    "     print_level = 0,"
+    "     max_iter = 100,"
+    "     dt = 1.0,"
+    "     steps = 1 "
+    "   }"
+    "}");
+
+  auto inlet = std::make_shared<axom::inlet::Inlet>(luaReader, dataStore.getRoot());
+
+  //
+  // Define input deck schema
+  //
+
+  inlet->addString("thermal_solver/mesh/filename", "filename");
```

Comment:
`thermal_solver/mesh` doesn't appear to be in the above lua string.  

If that's part of what you're testing for, it might help to add a comment.

Reply:
Done.